core/shared/src/main/scala/io/github/timwspence/cats/stm/STM.scala

```diff
@@ -47,49 +47,55 @@ object STM {
    * should be able to set the limit much higher and make use of all available
    * resources.
    */
-  def runtime[F[_]](n: Long)(implicit F: Async[F]): F[STM[F]] =
+  def runtime[F[_]](n: Long)(implicit F: Async[F]): F[STM[F]] = runtimeIn[F, F](n)
```

Comment:
Just curious why this is needed? (as I said on discord, I know _nothing_ about ScalaJS)

Reply:
So, on Scala.js there are no threads to block, and your ability to do something like `unsafeRunSync` depends on the effect type you're using:

- Safe: CE2-3 SyncIO, monix Coeval
- Does not compile: Dispatcher, CE3 IO, monix Task
- May throw runtime exception: CE2 IO

So, the test code using `IO#unsafeRunSync` simply wouldn't compile for SJS.

Dual-effect constructors, while being 100% pure, allow users to have a polymorphic "unsafe" factory by doing `runtimeIn[SyncIO, F](4).unsafeRunSync()` (and other reasons, check [this CE issue](https://github.com/typelevel/cats-effect/issues/373)). This would compile and always work as expected on JS, and that's what I changed the tests to do, but it might be generally helpful for users on JS side (where a significant number of interop cases is "impure code calls you" Soviet interop and in rare cases it's even "impure code calls your class' constructor").